### ISSUE TITLE
Does not work on Linux

### ISSUE THREAD

**Issue by pkese** [kevin-montrose/FeatherDotNet#3]
I'm seeing this problem:

    Unhandled Exception: System.InvalidOperationException: 
         Encoutered PlatformNotSupportedException trying to create a memory mapped proxy for passed bytes:   
         Named maps are not supported.  
         at FeatherDotNet.FeatherReader.ReadFromBytes(Byte[] bytes, BasisType basis) in C:\Users\kmontrose\Dropbox\FeatherDotNet\FeatherDotNet\FeatherReader.cs:line 120

Apparently Named memory mapped files are a *Windows-only* feature. On Linux there should be a filesystem entry or something used as a handle to share memory buffers around. See https://github.com/dotnet/corefx/issues/20091

I assume this would probably be reproducible on Windows by running tests inside the WSL Linux subsystem.

Maybe we could (for start) have an option to at least just load data into memory (i.e. without buffer sharing support).

**Comment by pkese** [comment]
Temporarily I have found the following workaround:  
- I have cloned the repository,
- made `FeatherReader.TryRead` a public method
- and then I call `TryRead` with a memory mapped file that I open myself.

**Comment by neroarth94** [comment]
Hello, may I know Is there any plan on fixing this issue in Linux?

**Comment by pkese** [comment]
This code dates back to 2018 and implements Feather format V1.

Apache Arrow foundation introduced new Feather format V2 in April 2020 so this is now a bit obsolete.
It is also unclear how much support there is for Feather format even within the Arrow foundation (they are referring to it as a format that was created as 'a proof of concept'): https://arrow.apache.org/docs/python/feather.html

You might have more luck checking at https://github.com/apache/arrow - there's a more complete C# implementation of Arrow concepts there. Even if there isn't any explicit Feather support in C#, maybe (?) they have an option to load Feather files into memory using their C++ code from some DLL and then access that Arrow memory chunk from C#.

**Comment by neroarth94** [comment]
How did you reproduce this issue in linux? because I was trying out this library on both windows 10 and linux (ubuntu 20.04) but I did not encountered any exceptions when writing and reading the feather files.

I was referring to the WriteTests.cs and have used only the FeatherWriter(temp, WriteMode.Eager) and FeatherReader.ReadFromFile(temp, BasisType.Zero) function but it seems to work fine for me. I saw that MemoryMappedFile is being used in the ReadFromFile function though.

This issue is fixed? or is it because MemoryMappedFile is supported in ubuntu but not on other linux platforms?
